tests/fixtures/cloud_uploader.py

```diff
@@ -128,8 +134,11 @@ def upload_file(
             *args,
             **kwargs) -> None:
         logger.info("%s", f"Uploading {local_path} to gs://{self.bucket.name}/{remote_path}")
+        fp = os.path.join(self.local_root, local_path)
         blob = self.bucket.blob(remote_path)
-        blob.upload_from_filename(os.path.join(self.local_root, local_path))
+        blob.upload_from_filename(fp)
         if metadata_keys:
```

Comment:
I think you have to make this unconditional.  If the caller doesn't specify metadata_keys, you should still set the size.

Reply:
Then size metadata will be included on manifests in fixtures/datafiles/*, which we may not want?

Reply:
Oh you are right. Hmmm, I’m not sure what the best way to proceed is. I forgot the manifests also get uploaded through this.

My preferred option is to add an optional flag to upload_file that turns on setting the size field, but I’m open to ideas.

Reply:
Mostly I'm agnostic... The other option seems to be including size when other fields are present, which involves fewer changes but is not explicit. Are there use cases for metadata fields *without* size?

Reply:
No, there is no use case for metadata fields without size.  Let's not dwell on this.  Document this behavior in `upload_file` and let's move on.